server/src/main/java/org/elasticsearch/action/admin/indices/create/AutoCreateAction.java

```diff
@@ -210,7 +213,7 @@ private CreateIndexClusterStateUpdateRequest buildSystemIndexUpdateRequest(Syste
                         updateRequest.aliases(Set.of(new Alias(aliasName)));
                     }
 
-                    logger.debug("Auto-creating system index {}", concreteIndexName);
+                    logger.debug("Auto-creating system index {}", indexName);
```

Comment:
I think it might be useful to call out in this log line when `targetIndex != indexName` and include both when that's the case.